### PR TITLE
docs - install file/dir locations; misc install-related edits

### PR DESCRIPTION
**Pull request by lisakowen**
install-related doc updates, including:
- new file layout
- relocating $PXF_BASE
- pxf [cluster] prepare
- PXF_CONF -> PXF_BASE
- templates now in $PXF_HOME/templates
- bundling a new version of postgresql jdbc jar

NOT IN THIS PR:  info about specific configuration/logging changes (for example, this PR includes a change from PXF_CONF ->PXF_BASE for a catalina.out file location.  this reference will be removed in a future PR.)

doc review site links:
- files/directories, relocate topic:  http://docs-lisa-pxf6-instmisc.cfapps.io/pxf/6-0/using/about_pxf_dir.html
- pxf cluster ref page:  http://docs-lisa-pxf6-instmisc.cfapps.io/pxf/6-0/using/ref/pxf-cluster.html
- pxf ref page: http://docs-lisa-pxf6-instmisc.cfapps.io/pxf/6-0/using/ref/pxf.html

### CODE REVIEW

File: docs/content/about_pxf_dir.html.md.erb [hunk 1]
```diff
@@ -2,45 +2,54 @@
...
+When you install PXF 6, the default `$PXF_BASE` is `$PXF_HOME`.
 
 ## <a id="installed"></a>PXF Installation Directories
 
-The following PXF files and directories are installed to `$PXF_HOME` when you install Greenplum Database or the PXF `rpm` or `deb`:
+The following PXF files and directories are installed to `$PXF_HOME` when you install Greenplum Database or the PXF 6 `rpm` or `deb` package:
+
+| Directory | Description                                                                                                   |
+|--------------------------------|------------------------------------------------------------------------------------------|
+| application/                | The PXF Spring Boot application JAR file.   |
```
Comment (dyozie) [inline docs/content/about_pxf_dir.html.md.erb:17]: Is it useful to the customer to qualify this with "Spring Boot"?  Just wondering how much of that is necessary to describe.

File: docs/content/about_pxf_dir.html.md.erb [hunk 2]
```diff
@@ -2,45 +2,54 @@
...
+| bin/                | The PXF command line executable directory.   |
+| commit.sha          | The commit identifier for this PXF release.   |
+| gpextable/           | The PXF extension files. PXF copies the `pxf.control` file in this directory to the Greenplum installation (`$GPHOME`) on a single host or all hosts in the cluster when you run the `pxf [cluster] register` command.   |
```
Comment (dyozie) [inline docs/content/about_pxf_dir.html.md.erb:20]: host or all -> host or on all

File: docs/content/about_pxf_dir.html.md.erb [hunk 3]
```diff
@@ -2,45 +2,54 @@
...
+| share/                | The directory for shared PXF files that you may require depending on the external data stores that you access. `share/` currently includes only the PXF HBase JAR file. |
```
Comment (dyozie) [inline docs/content/about_pxf_dir.html.md.erb:21]: currently -> initially

File: docs/content/about_pxf_dir.html.md.erb [hunk 4]
```diff
@@ -2,45 +2,54 @@
...
+| conf/                | The location of user-customizable PXF configuration files for PXF runtime and logging configuration settings. This directory contains the `pxf-application.properties`, `pxf-env.sh`, `pxf-log4j2.xml`, and `pxf-profiles.xml` files. |
+| keytabs/  | The default location for the PXF service Kerberos principal keytab file.  The `keytabs/` directory and contained files are readable only by the `gpadmin` user. |
+| lib/     | The default PXF directory for user-added runtime JAR dependencies. The `native/` subdirectory is the default PXF runtime directory for native libraries. |
+| logs/                | The PXF runtime log file directory. The `logs/` directory and log files are readable only by the `gpadmin` user. |
+| run/         | The default PXF run directory. After starting PXF, contains a PXF process id file. The default `$PXF_OOM_DUMP_PATH` resides in a `pxf_heap_dump/` subdirectory. `run/` and contained files and directories are readable only by the `gpadmin` user. |
```
Comment (dyozie) [inline docs/content/about_pxf_dir.html.md.erb:33]: contains -> this directory contains

**Review by dyozie — APPROVED** [review]

**Review by ashuka24 — APPROVED** [review]

File: docs/content/about_pxf_dir.html.md.erb [hunk 5]
```diff
@@ -2,45 +2,54 @@
...
+The following PXF files and directories are installed to `$PXF_HOME` when you install Greenplum Database or the PXF 6 `rpm` or `deb` package:
+
+| Directory | Description                                                                                                   |
+|--------------------------------|------------------------------------------------------------------------------------------|
+| application/                | The PXF Spring Boot application JAR file.   |
+| bin/                | The PXF command line executable directory.   |
+| commit.sha          | The commit identifier for this PXF release.   |
+| gpextable/           | The PXF extension files. PXF copies the `pxf.control` file in this directory to the Greenplum installation (`$GPHOME`) on a single host or on all hosts in the cluster when you run the `pxf [cluster] register` command.   |
+| share/                | The directory for shared PXF files that you may require depending on the external data stores that you access. `share/` initially includes only the PXF HBase JAR file. |
+| templates/  | The PXF directory for template server configuration files. |
```
Comment (denalex) [inline docs/content/about_pxf_dir.html.md.erb:22]:
```suggestion
| templates/  | The PXF directory for server configuration file templates. |
```

File: docs/content/about_pxf_dir.html.md.erb [hunk 6]
```diff
@@ -2,45 +2,54 @@
...
+| version              | The PXF version.             |
+
+The following PXF directories are installed to `$PXF_BASE` when you install Greenplum Database or the PXF 6 `rpm` or `deb` package:
```
Comment (denalex) [inline docs/content/about_pxf_dir.html.md.erb:25]:
```suggestion
The following PXF directories are installed to `$PXF_BASE` when you install Greenplum Database or the PXF 6.x `rpm` or `deb` package:
```

File: docs/content/about_pxf_dir.html.md.erb [hunk 7]
```diff
@@ -2,45 +2,54 @@
...
-| lib/                 | The PXF library directory.                                                                                                                                                                                                                       |
-| templates/                | Configuration templates for PXF. |
-| version              | The PXF version.                                                                                                                                                                                                                       |
+| conf/                | The location of user-customizable PXF configuration files for PXF runtime and logging configuration settings. This directory contains the `pxf-application.properties`, `pxf-env.sh`, `pxf-log4j2.xml`, and `pxf-profiles.xml` files. |
+| keytabs/  | The default location for the PXF service Kerberos principal keytab file.  The `keytabs/` directory and contained files are readable only by the `gpadmin` user. |
```
Comment (denalex) [inline docs/content/about_pxf_dir.html.md.erb:30]: the installation user does not have to be `gpadmin`

Comment (denalex) [inline docs/content/about_pxf_dir.html.md.erb:30]: above we say `the location of ...`, here `the location for ...` -- should probably use just one consistent form

File: docs/content/about_pxf_dir.html.md.erb [hunk 8]
```diff
@@ -2,45 +2,54 @@
...
+| lib/     | The default PXF directory for user-added runtime JAR dependencies. The `native/` subdirectory is the default PXF runtime directory for native libraries. |
```
Comment (denalex) [inline docs/content/about_pxf_dir.html.md.erb:31]:
```suggestion
| lib/     | The location for user-added runtime JAR dependencies. The `native/` subdirectory is the default PXF runtime directory for native libraries. |
```

File: docs/content/about_pxf_dir.html.md.erb [hunk 9]
```diff
@@ -2,45 +2,54 @@
...
+| logs/                | The PXF runtime log file directory. The `logs/` directory and log files are readable only by the `gpadmin` user. |
+| run/         | The default PXF run directory. After starting PXF, this directory contains a PXF process id file. The default `$PXF_OOM_DUMP_PATH` resides in a `pxf_heap_dump/` subdirectory. `run/` and contained files and directories are readable only by the `gpadmin` user. |
```
Comment (denalex) [inline docs/content/about_pxf_dir.html.md.erb:33]:
```suggestion
| run/         | The default PXF run directory. After starting PXF, this directory contains a PXF process id file `pxf-app.pid`. The default `$PXF_OOM_DUMP_PATH` resides in a `pxf_heap_dump/` subdirectory. `run/` and contained files and directories are readable only by the `gpadmin` user. |
```

File: docs/content/about_pxf_dir.html.md.erb [hunk 10]
```diff
@@ -2,45 +2,54 @@
...
+| servers/  | The configuration directory for PXF servers; each subdirectory contains a server definition, and the name of the subdirectory identifies the name of the server. The default server is named `default`. The Greenplum Database administrator may configure other servers. |
 
+Refer to [Configuring PXF](instcfg_pxf.html) and [Starting PXF](cfginitstart_pxf.html#start_pxf) for detailed information about the PXF configuration and startup commands and procedures.
 
-## <a id="runtime"></a>PXF Runtime Directories
+## <a id="movebase"></a>Relocating $PXF_BASE
 
-During initialization and startup, PXF creates the following internal directories in `$PXF_HOME`:
+If you require that `$PXF_BASE` reside in a directory distinct from `$PXF_HOME`, you can change it from the default location to a location of your choosing after you install PXF 6.
```
Comment (denalex) [inline docs/content/about_pxf_dir.html.md.erb:40]:
```suggestion
If you require that `$PXF_BASE` reside in a directory distinct from `$PXF_HOME`, you can change it from the default location to a location of your choosing after you install PXF 6.x.
```

File: docs/content/pxf_kerbhdfs.html.md.erb [hunk 11]
```diff
@@ -163,11 +163,11 @@ When you configure PXF for secure HDFS using an AD Kerberos KDC server, you will
     gpadmin@master$ pxf cluster restart
     ```
 
-6. Step 7 does not synchronize the keytabs in `$PXF_CONF`. You must distribute the keytab file to `$PXF_CONF/keytabs/`. Locate the keytab file, copy the file to the `$PXF_CONF` user configuration directory, and set required permissions. For example:
+6. Step 7 does not synchronize the keytabs in `$PXF_BASE`. You must distribute the keytab file to `$PXF_BASE/keytabs/`. Locate the keytab file, copy the file to the `$PXF_BASE` user configuration directory, and set required permissions. For example:
```
Comment (denalex) [inline docs/content/pxf_kerbhdfs.html.md.erb:166]: which step 7 ? if we are talking about `pxf cluster sync` then it should be step 5 above.

Comment (denalex) [inline docs/content/pxf_kerbhdfs.html.md.erb:166]:
```suggestion
6. Step 7 does not synchronize the keytabs in `$PXF_BASE`. You must distribute the keytab file to `$PXF_BASE/keytabs/`. Locate the keytab file, copy the file to the `$PXF_BASE` runtime configuration directory, and set required permissions. For example:
```

File: docs/content/ref/pxf.html.md.erb [hunk 12]
```diff
@@ -34,9 +35,9 @@ You can initialize or reset PXF on the master, master standby, or a specific seg
```
Comment (denalex) [inline docs/content/ref/pxf.html.md.erb:35]: ^^^ `You can initialize or reset PXF on the master ...` reset is no longer doing anything and we do not need init either.

File: docs/content/ref/pxf.html.md.erb [hunk 13]
```diff
@@ -34,9 +35,9 @@ You can initialize or reset PXF on the master, master standby, or a specific seg
 
 You can start, stop, or restart the PXF service instance on a specific segment host, or display the status of the PXF service instance running on a segment host.
 
-You can copy the extension files from a PXF installation on the host to the Greenplum installation on the host after a Greenplum upgrade.
+You can prepare a new `$PXF_BASE` runtime configuration directory on the host, or copy the extension files from a PXF installation on the host to the Greenplum installation on the host after a Greenplum upgrade.
```
Comment (denalex) [inline docs/content/ref/pxf.html.md.erb:38]: we now copy only extension control file , not "extension files"

File: docs/content/ref/pxf.html.md.erb [hunk 14]
```diff
@@ -34,9 +35,9 @@ You can initialize or reset PXF on the master, master standby, or a specific seg
...
 
-(Use the [`pxf cluster`](pxf-cluster.html#topic1) command to copy extension files to, initialize, or reset PXF on all hosts, synchronize the PXF configuration to the Greenplum Database cluster, or to start, stop, or display the status of the PXF service instance on all segment hosts in the cluster.)
+(Use the [`pxf cluster`](pxf-cluster.html#topic1) command to prepare a new `$PXF_BASE`, copy extension files to, initialize, or reset PXF on all hosts, synchronize the PXF configuration to the Greenplum Database cluster, or to start, stop, or display the status of the PXF service instance on all segment hosts in the cluster.)
```
Comment (denalex) [inline docs/content/ref/pxf.html.md.erb:40]:
```suggestion
(Use the [`pxf cluster`](pxf-cluster.html#topic1) command to prepare a new `$PXF_BASE`, copy extension file to `$GPHOME` on all hosts, synchronize the PXF configuration to the Greenplum Database cluster, or to start, stop, or display the status of the PXF service instance on all segment hosts in the cluster.)
```

File: docs/content/ref/pxf.html.md.erb [hunk 15]
```diff
@@ -49,6 +50,9 @@ You can copy the extension files from a PXF installation on the host to the Gree
 <dt>init</dt>
 <dd>Initialize the PXF service instance on the host. The command also copies the PXF extension files from the PXF installation to the Greenplum installation on the host. (Specify the `--skip-register` option to instruct PXF to skip this initialization step.) When you initialize PXF, you must identify the PXF user configuration directory via an environment variable named `$PXF_CONF`. If you do not set `$PXF_CONF` prior to initializing PXF, PXF prompts you to accept or decline the default user configuration directory, `$HOME/pxf`, during the initialization process. See [Options](pxf.html#options).</dd>
 
+<dt>prepare</dt>
+<dd>Prepare a new `$PXF_BASE` directory on the host. When you run the command, you must identify the new PXF runtime configuration directory via an environment variable named `PXF_BASE`. PXF copies template runtime configuration files and directories to this `$PXF_BASE`.</dd>
```
Comment (denalex) [inline docs/content/ref/pxf.html.md.erb:54]:
```suggestion
<dd>Prepare a new `$PXF_BASE` directory on the host. When you run the command, you must identify the new PXF runtime configuration directory via an environment variable named `PXF_BASE`. PXF copies runtime configuration file templates and directories to this `$PXF_BASE`.</dd>
```

File: docs/content/ref/pxf.html.md.erb [hunk 16]
```diff
@@ -49,6 +50,9 @@ You can copy the extension files from a PXF installation on the host to the Gree
...
+
 <dt>register</dt>
 <dd>Copy the PXF extension files from the PXF installation on the host to the Greenplum installation on the host. This command requires that `$GPHOME` be set, and is typically run after you upgrade your Greenplum Database installation.</dd>
```
Comment (denalex) [inline docs/content/ref/pxf.html.md.erb:57]: only 1 file is now copied.

File: docs/content/s3_objstore_cfg.html.md.erb [hunk 17]
```diff
@@ -10,12 +10,12 @@ You can use PXF to access S3-compatible object stores. This topic describes how
 
 To access data in an object store, you must provide a server location and client credentials. When you configure a PXF object store connector, you add at least one named PXF server configuration for the connector as described in [Configuring PXF Servers](cfg_server.html).
 
-PXF provides a template configuration file for each object store connector. These template files are located in the `$PXF_CONF/templates/` directory.
+PXF provides a template configuration file for each object store connector. These template files are located in the `$PXF_HOME/templates/` directory.
```
Comment (denalex) [inline docs/content/s3_objstore_cfg.html.md.erb:13]:
```suggestion
PXF provides a configuration file template for each object store connector. These template files are located in the `$PXF_HOME/templates/` directory.
```

File: docs/content/about_pxf_dir.html.md.erb [hunk 18]
```diff
@@ -2,45 +2,54 @@
 title: About the PXF Installation and Configuration Directories
 ---
 
-PXF is installed to `$GPHOME/pxf` on your master and segment nodes when you install Greenplum Database. If you install the PXF `rpm` or `deb` package, PXF is installed to `/usr/local/pxf-gp<greenplum-major-version>`, or to a directory of your choosing (CentOS/RHEL only). This documentation uses `$PXF_HOME` to refer to the PXF installation directory.
+PXF may be installed to `$GPHOME/pxf` on your master and segment nodes when you install Greenplum Database. If you install the PXF `rpm` or `deb` package, PXF is installed to `/usr/local/pxf-gp<greenplum-major-version>`, or to a directory of your choosing (CentOS/RHEL only). This documentation uses `$PXF_HOME` to refer to the PXF installation directory.
```
Comment (bradfordb-vmware) [inline docs/content/about_pxf_dir.html.md.erb:5]: The [Fedora packaging guidelines](https://docs.fedoraproject.org/en-US/packaging-guidelines/#_relocatable_packages) discourages the use of relocatable RPMs. I understand that we have some users the require this and that's why we use it. Would it make sense to separate the issue of relocating to a separate section for advanced topics?

File: docs/content/about_pxf_dir.html.md.erb (hunk 1 above)
Comment (bradfordb-vmware) [reply docs/content/about_pxf_dir.html.md.erb:17]: What about `PXF server application`?

File: docs/content/ref/pxf-cluster.html.md.erb [hunk 19]
```diff
@@ -15,6 +15,7 @@ where `<command>` is:
 ``` pre
 help
 init
+prepare
```
Comment (frankgh) [inline docs/content/ref/pxf-cluster.html.md.erb:18]: I assume pxf migrate will be documented later.

File: docs/content/troubleshooting_pxf.html.md.erb [hunk 20]
```diff
@@ -146,7 +146,7 @@ By default, PXF is configured such that when the PXF JVM detects an out of memor
 
 When auto-kill is enabled and the PXF JVM detects an OOM condition and kills the PXF server on the segment host:
 
-- PXF logs the following messages to `$PXF_CONF/logs/catalina.out` on the segment host:
+- PXF logs the following messages to `$PXF_BASE/logs/catalina.out` on the segment host:
```
Comment (frankgh) [inline docs/content/troubleshooting_pxf.html.md.erb:149]: we have a single log file now, it's called `pxf-service.log` by default. maybe out of scope of this PR though

File: docs/content/ref/pxf.html.md.erb (hunk 12 above)
Comment (lisakowen) [reply docs/content/ref/pxf.html.md.erb:35]: the next PR i submit will take care of init/reset references and deprecation.

File: docs/content/ref/pxf.html.md.erb (hunk 13 above)
Comment (lisakowen) [reply docs/content/ref/pxf.html.md.erb:38]: the next PR also will include updates to the register command (copies only one file now, only need to do it once.)

File: docs/content/ref/pxf-cluster.html.md.erb (hunk 19 above)
Comment (lisakowen) [reply docs/content/ref/pxf-cluster.html.md.erb:18]: yes, will document pxf migrate in a later PR.

File: docs/content/troubleshooting_pxf.html.md.erb (hunk 20 above)
Comment (lisakowen) [reply docs/content/troubleshooting_pxf.html.md.erb:149]: will address log-related changes in a separate PR.

File: docs/content/pxf_kerbhdfs.html.md.erb (hunk 11 above)
Comment (lisakowen) [reply docs/content/pxf_kerbhdfs.html.md.erb]: yes, the cluster sync step, and it is actually step 7.  (it doesn't matter what the list numbers are, markdown will render/number them correctly.)

File: docs/content/about_pxf_dir.html.md.erb [hunk 21]
```diff
@@ -2,45 +2,54 @@
...
+
+In PXF 5.x, you specified a `$PXF_CONF` directory for configuration and runtime files when you initialized PXF.  In PXF 6.x, configuration and runtime files are located in `$PXF_BASE`, and no initialization is required.
```
[the remark this answers is not in the document]
Comment (lisakowen) [reply docs/content/about_pxf_dir.html.md.erb]: @denalex - i've added more information about PXF_BASE on the intro page.  i've kept the info about the prepare command and setting the environment variable in the "relocating PXF_BASE" topic.

re: terminology, what is the best way to identify the non-PXF_BASE contents of PXF_HOME?  i think we've standardized on "runtime configuration files/directories" when describing PXF_BASE.

File: docs/content/about_pxf_dir.html.md.erb (hunk 18 above)
Comment (lisakowen) [reply docs/content/about_pxf_dir.html.md.erb:5]: @bradfordb-vmware - can we defer this discussion and possible doc changes to another PR?  this is one of the few places that we mention relocatable rpm.  most of the content is in the install topic.  we can talk about moving it out of the main install procedure.

**Review by denalex — APPROVED** [review]

File: docs/content/about_pxf_dir.html.md.erb (hunk 18 above)
[the remark this answers is not in the document]
Comment (lisakowen) [reply docs/content/about_pxf_dir.html.md.erb:5]: @frankgh - yes, if pxf 6 is embedded in greenplum, with the new rpm file layout/PXF_BASE we are back to the same issues that we had with pxf 5.x and greenplum upgrades - > upgrading greenplum could munge the runtime configuration dirs/files.  seems like there are a few options:
1. remove pxf from greenplum 6 distribution
2. keep pxf 5.15.1 in greenplum 6 distribution, no further pxf-integrated updates
3. include pxf 5.latest (5.16) in the greenplum distribution, no further pxf-integrated updates
4. include pxf 6.latest in the greenplum distribution, [possibly update with newer versions of pxf 6.x.]

1, 2, and 3 require that the user must install the pxf rpm to get pxf 6.  if we limit it to one of these options, we won't need to worry about greenplum upgrades with pxf 6.  if we choose 4, we will probably want to recommend that the user relocate their PXF base on their first use of pxf 6.

**Comment by lisakowen** [comment]
thank you for reviewing @denalex , @ashuka24 , @bradfordb-vmware , @frankgh !

**Pull request merged**